collect_app/src/main/java/org/odk/collect/android/formmanagement/InstancesAppState.kt

```diff
@@ -1,28 +1,29 @@
 package org.odk.collect.android.formmanagement
 
+import android.content.Context
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
+import org.odk.collect.android.provider.InstanceProviderAPI
 import org.odk.collect.android.utilities.InstancesRepositoryProvider
 import org.odk.collect.forms.instances.Instance
 import org.odk.collect.forms.instances.InstancesRepository
 import javax.inject.Singleton
 
 /**
- * Stores reactive state of current instances count with various different statuses. This (as a
- * singleton) can be read or updated by different parts of the app without needing reactive data
- * in the [InstancesRepository].
+ * Stores reactive state of instances. This (as a singleton) can be read or updated by
+ * different parts of the app without needing reactive data in the [InstancesRepository].
  */
 @Singleton
-class InstancesCountRepository {
+class InstancesAppState(private val context: Context) {
```

Comment:
Trying this naming on for size. In my head there is potentially a pattern where all our interactions from "Android" code go through state objects like this (you could imagine this could have a `instanceSubmitted` or a `finalizeInstance` method). That would basically give us a Redux-esque state system where these app state objects could expose both the state and the event/reducers. But you know, lets just hold off on reinventing that wheel for the moment.

Reply:
Like it.